### DISCUSSION TITLE
Compatible with OpenLayers 10

### DISCUSSION THREAD

**Discussion by levelost** [openlayers/ol-mapbox-style#1181] [General]
Is the library compatible with OpenLayers 10?

**Comment by ahocevar** [comment]
Yes, what makes you think otherwise?

**Comment by levelost** [comment]
But MapboxVectorLayer getSource() returns RenderFeature. RenderFeature ≠ Source.

**Comment by ahocevar** [comment]
Not sure I understand what you mean. Can you provide a code snippet that helps me understand?

**Comment by levelost** [comment]
```ts
applyStyle(this, this.baseUrl.replace(layersPattern, layers.toString()), {})
.then(() => {
    this.getSource()?.refresh();
    this.setVisible(true);
});
```

> Property 'refresh' does not exist on type 'RenderFeature'.

**Comment by levelost** [comment]
```ts
this.setSource(
    new VectorTile({
        projection: map.getView().getProjection(),
        format: new MVT()
    })
);
```

> Argument of type 'VectorTile<RenderFeature>' is not assignable to parameter of type 'RenderFeature'.

**Reply by ahocevar** [reply]
What type is `this` in your code snippets?

**Reply by levelost** [reply]
```ts
class MyLayer extends MapboxVectorLayer{
    //...
    myFunction() {
        this.setSource(
            new VectorTile({
                projection: map.getView().getProjection(),
                format: new MVT()
            })
        );
    }
}
```

**Reply by ahocevar** [reply]
Ok, I see what the problem is: `MapboxVectorLayer` uses the type generics that were only used in v9.2 - not before and not after. I'm going to publish a fix for that.

**Reply by ahocevar** [reply]
#1184
